streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -55,12 +60,29 @@
     private static final String APP_ID = "app-id";
 
     private final KStreamBuilder builder = new KStreamBuilder();
-    @Rule
-    public final KStreamTestDriver driver = new KStreamTestDriver();
+    private final ConsumerRecordFactory<String, String> recordFactory = new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer());
+    private TopologyTestDriver driver;
+    private Properties props;
 
     @Before
-    public void setUp() {
+    public void setup() {
         builder.setApplicationId(APP_ID);
+        Properties props = new Properties();
```

Comment:
Why do we need this variable? Can't we use `this.props` directly?

Reply:
I use it just to instantiate a `Properties` in `@Before`, which I then assign to `this.props`, in order to avoid having to do it in each test, because the same properties are used for all tests. Should I do it in the tests instead?

Reply:
Doing in `@Before` is fine. But we don't need to call `new` each time. `this.props` will be a new empty `Properties` instance anyway -- we don't need the second object.

Reply:
If it's the same for every test, you could also just create a `getProperties()` method for the tests to call when they need the properties. This way, you won't need a field to store the properties in.

Reply:
I now use `this.props` directly in all test classes. The default values for the test are loaded in `@Before`, and I clean them up in with `props.clear()` in  `@After`.